beacon-chain/rpc/validator/sync_committee.go

```diff
@@ -0,0 +1,65 @@
+package validator
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// GetSyncMessageBlockRoot retrieves the sync committee block root of the beacon chain.
+func (vs *Server) GetSyncMessageBlockRoot(ctx context.Context, req *ethpb.SyncMessageBlockRootRequest) (*ethpb.SyncMessageBlockRootResponse, error) {
+	// Short cut, where copying state and processing slots are not required.
+	if req.Slot-1 == vs.HeadFetcher.HeadSlot() {
```

Comment:
Lets handle overflows here in the event `slot` is 0

Reply:
Actually why not make the `req.Slot` denote `currentSlot - 1` ? Instead of doing all the checks here.